src/System.Linq.Expressions/src/System/Runtime/CompilerServices/ReadOnlyCollectionBuilder.cs

```diff
@@ -544,24 +583,27 @@ object IEnumerator.Current
 
             public bool MoveNext()
             {
-                if (_version == _builder._version)
+                ReadOnlyCollectionBuilder<T> localBuilder = _builder;
```

Comment:
Yep, which might have been intentional or an oversight on @VSadov's side (he marked is up-for-grabs). I agree that in general it would be great if we clarify these things before marking things as "up for grabs", on the other hand, I don't have expectation we will be 100% perfect all the time. That's why we have the safety net of contributor's own judgement and critical thinking and code reviews as the last line of defense.